### PR TITLE
Fixes bug when trying to use local ip address as host name.

### PR DESCRIPTION
**Pull request by KieranDevvs**
Currently if you use the `ISMBClient.Connect()` overload where it takes in a string as the server name, and you pass it an ip address rather than a domain name, it fails to resolve the IP address due to `System.Net.DNS` resolving the hostname of the local IP address, but does not add the ip address into the address list.

This solution will get the IP address for a domain name or an IP address.

### CODE REVIEW

File: SMBLibrary/Client/SMB2Client.cs [hunk 1]
```diff
@@ -66,12 +66,12 @@ public SMB2Client()
         public bool Connect(string serverName, SMBTransportType transport)
         {
             m_serverName = serverName;
-            IPHostEntry hostEntry = Dns.GetHostEntry(serverName);
-            if (hostEntry.AddressList.Length == 0)
+            IPAddress[] hostAddresses = Dns.GetHostAddresses(serverName);
+            if (hostAddresses.Length == 0)
             {
-                throw new Exception(String.Format("Cannot resolve host name {0} to an IP address", serverName));
+                throw new Exception(string.Format("Cannot resolve host name {0} to an IP address", serverName));
```
Comment (TalAloni) [inline SMBLibrary/Client/SMB2Client.cs:72]: This is an unrelated change and should not be part of this pull request

**Comment by TalAloni** [comment]
Thanks Kieran,
I understand where this change might be useful.
I found it best to avoid combining unrelated changes and have each pull request focus on one change.

File: .gitignore [hunk 2]
```diff
@@ -229,3 +229,4 @@ $RECYCLE.BIN/
 
 # Windows shortcuts
 *.lnk
+/.vs
```
[the remark this answers is not in the document]
Comment (KieranDevvs) [reply .gitignore:232]: I agree this should be part of a separate PR. Ill remove this.

File: SMBLibrary/Client/SMB1Client.cs [hunk 3]
```diff
@@ -60,12 +60,12 @@ public SMB1Client()
 
         public bool Connect(string serverName, SMBTransportType transport)
         {
-            IPHostEntry hostEntry = Dns.GetHostEntry(serverName);
-            if (hostEntry.AddressList.Length == 0)
+            IPAddress[] hostAddresses = Dns.GetHostAddresses(serverName);
+            if (hostAddresses.Length == 0)
             {
-                throw new Exception(String.Format("Cannot resolve host name {0} to an IP address", serverName));
+                throw new Exception(string.Format("Cannot resolve host name {0} to an IP address", serverName));
```
[the remark this answers is not in the document]
Comment (KieranDevvs) [reply SMBLibrary/Client/SMB1Client.cs:66]: You don't have any contribution guidelines for this project so its unclear what the scope of refactorization sits at?
I was seeing this as "I'm changing method X, thus I should leave method X in a better state than I found it".
Or are you expecting one big PR for refactorization's and code cleanup?

[the remark this answers is not in the document]
Comment (TalAloni) [reply SMBLibrary/Client/SMB1Client.cs:66]: I'm expecting a pull request to focus on one change and not stray into improving unrelated things in the same PR.
